src/item_location.cpp

```diff
@@ -344,37 +398,74 @@ item_location::item_location( const vehicle_cursor &vc, item *which )
 
 bool item_location::operator==( const item_location &rhs ) const
 {
-    return ( ptr ? ptr->what : nullptr ) == ( rhs.ptr ? rhs.ptr->what : nullptr );
+    return ( ptr ? ptr->target() : nullptr ) == ( rhs.ptr ? rhs.ptr->target() : nullptr );
 }
 
 bool item_location::operator!=( const item_location &rhs ) const
 {
-    return ( ptr ? ptr->what : nullptr ) != ( rhs.ptr ? rhs.ptr->what : nullptr );
+    return ( ptr ? ptr->target() : nullptr ) != ( rhs.ptr ? rhs.ptr->target() : nullptr );
```

Comment:
This `ptr ? ptr->target() : nullptr` idiom is repeated a lot.
It would be cleaner to add a `item_nowhere` class or something like that that always returns `nullptr`.